### ISSUE TITLE
Persisting JS Maps not supported

### ISSUE THREAD

**Issue by MaRaSu** [quarrant/mobx-persist-store#35]
MobX supports Javascript Maps and especially when lacking support for proxies (like in React Native without Hermes engine) there are some reasons to use Maps instead of plain objects. 

However Maps are not supported by JSON.stringify. There are a number of workarounds for that. But none of these work with mobx-persist-store since JSON.parse & stringify are hidden inside the library instead of being exposed in the StorageAdapter interface.

**Comment by quarrant** [comment]
mobx-persist-store works with Maps. Can you show your code?

**Comment by MaRaSu** [comment]
@quarrant indeed the code actually seems to be supporting it. I was too quick to comment it relying just on JSON.stringify.

Anyhow persisting Maps was not working for me so I already switched to plain objects with some tricks to make them as observable as Maps. Come to think of this problem I do have a unusual setup as I'm using expo for react native and mobx babel config for class properties has a conflict with expo so I had to omit part of the config - so this issue might be as well related to mobx.

Let me see if I have time to put together something simple that could help in reproducing this while also letting to find out if this is related to problem with my babel config.

I think this issue can be closed until I can come back with more info.
